lib/worker/OperationQueue.js

```diff
@@ -0,0 +1,363 @@
+/*!
+ * Copyright (c) 2017-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const _continuityConstants = require('../continuityConstants');
+const _util = require('../util');
+const database = require('bedrock-mongodb');
+const logger = require('../logger');
+const pLimit = require('p-limit');
+
+// Controls the max concurrency for a given number of promises executing. This
+// is done to ensure we reduce pressure on the NodeJS Promise Queue and limit
+// the number concurrent I/O operations. The number 25 was arbitrarily picked.
+const limit = pLimit(25);
+// Controls the maximum number of documents to pull from the database and cache
+// locally. The Math.max(DEFAULT_MAX_CACHED_QUEUE_SIZE, MAX_OPS_PER_REG_EVENT)
+// will be the final value used to limit the number of documents pulled from
+// the database.
+const DEFAULT_MAX_CACHED_QUEUE_SIZE = 250;
+
+// Error Messages
+const OPERATIONS_REMOVAL_ERROR = 'Failed to remove operations from queue.';
+const OPERATION_VALIDATION_ERROR = 'Non-critical operation validation error. ' +
+  'Invalid head of operation queue.';
+
+module.exports = class OperationQueue {
+  constructor({worker}) {
+    // continuity worker
```

Comment:
I'd rather us talk about it being the consensus worker -- or the session worker than talking about continuity as that's the name of just the low-level consensus algorithm itself. The worker runs the pipeline, does gossip, creates merge events and blocks ... all that.

```suggestion
    // session worker
```